Build failure: sweethome3d

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *sweethome3d.application*

### Build log

```
error: builder for '/nix/store/facj2cxp9klnjka42sx63kmsrgj0aimy-sweethome3d-application-7.2.drv' failed with exit code 1;
       last 10 log lines:
       >     [javac] /build/SweetHome3D-7.2-src/build.xml:68: warning: 'includeantruntime' was not set, defaulting to build.sysclasspath=last; set to false for repeatable builds
       >     [javac] Compiling 2 source files to /build/SweetHome3D-7.2-src/build/classes
       >     [javac] warning: [options] bootstrap class path not set in conjunction with -source 7
       >     [javac] error: Source option 7 is no longer supported. Use 8 or later.
       >     [javac] error: Target option 7 is no longer supported. Use 8 or later.
       >
       > BUILD FAILED
       > /build/SweetHome3D-7.2-src/build.xml:68: Compile failed; see the compiler error output for details.
       >
       > Total time: 0 seconds
       For full logs, run 'nix log /nix/store/facj2cxp9klnjka42sx63kmsrgj0aimy-sweethome3d-application-7.2.drv'.
```

### Notify maintainers

@edwtjo 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.18, NixOS, 24.05 (Uakari), 24.05.20240309.3030f18`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos, nixos-hardware"`
 - nixpkgs: `/home/sam/.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/misc/sweethome3d/default.nix
+++ b/pkgs/applications/misc/sweethome3d/default.nix
@@ -53,6 +53,9 @@ let
     nativeBuildInputs = [ makeWrapper unzip autoPatchelfHook ];
     buildInputs = [ ant jdk p7zip gtk3 gsettings-desktop-schemas libXxf86vm ];
 
+    # upstream targets Java 7 by default
+    env.ANT_ARGS = "-DappletClassSource=8 -DappletClassTarget=8 -DclassSource=8 -DclassTarget=8";
+
     buildPhase = ''
       runHook preBuild
 

--- a/pkgs/applications/misc/sweethome3d/editors.nix
+++ b/pkgs/applications/misc/sweethome3d/editors.nix
@@ -44,6 +44,9 @@ let
     nativeBuildInputs = [ makeWrapper unzip ];
     buildInputs = [ ant jdk gtk3 gsettings-desktop-schemas ];
 
+    # upstream targets Java 7 by default
+    env.ANT_ARGS = "-DappletClassSource=8 -DappletClassTarget=8 -DclassSource=8 -DclassTarget=8";
+
     postPatch = ''
       sed -i -e 's,../SweetHome3D,${applicationSrc},g' build.xml
       sed -i -e 's,lib/macosx/java3d-1.6/jogl-all.jar,lib/java3d-1.6/jogl-all.jar,g' build.xml

